roxie/roxiemem/roxiemem.cpp

```diff
@@ -6490,6 +6503,16 @@ bool DataBuffer::attachToRowMgr(IRowManager *rowMgr)
     }
 }
 
+#ifdef TRACK_DATABUFF_STATE
+static constexpr const char * stateNames[(unsigned)DBState::max] = { "zero", "unowned", "queued", "attached", "freed" };
+void DataBuffer::changeState(DBState oldState, DBState newState, const char * func)
+{
+    if (state != oldState)
```

Comment:
perhaps add a check for memTraceInconsistencies == true here just as a way to disable logging at runtime if ever desired ?